.github/workflows/tests.yml

```diff
@@ -38,18 +40,30 @@ jobs:
   build-snap:
     name: Build snap
     runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft }}
     needs: lint-unittests-docs
+    outputs:
+      files-changed: ${{ steps.files-changed.outputs.changed }}
     steps:
     - uses: actions/checkout@v2
     - uses: snapcore/action-build@v1
       id: snapcraft
     - name: Install snap
       run: sudo snap install --dangerous ${{ steps.snapcraft.outputs.snap }}
+    - name: Track changed files in the PR
+      id: files-changed
+      run: |
+        git remote add upstream https://github.com/canonical/juju-verify
+        git fetch upstream master
+        failed=0
+        git diff --name-only upstream/master | grep -qE '^(juju_verify|tests)' || failed=1
+        echo "::set-output name=changed::$failed"
```

Comment:
I also think that the "Build snap" could be conditioned by changes in (juju_verify|tests)?

Reply:
Creating a new job means spending more resources from the hosted runners env (from GH, since a new container would be deployed).

Creating a new step within an already existing job will make tests faster and reduce the (limited) amount of hours we can run CI jobs.

ack to conditioning the `Build snap` job to changes in those folders (and the snap folder). However, the previous job runs multiple tests in parallel, and in order to run `git-diff`, we would probably need an initial step in the `Build snap ` job. Would there be any other approach where the `git-diff` step is run before needed to deploy the env for the `Build snap` job?

Reply:
Ok that is true, we can move it into `lint-unittests-docs` with condition like this:
```yaml
- name: Verify package
  run: tox -e build-verify
- name: Track changed files in the PR
  if: ${{ matrix.python-version == 3.9 }}
  id: files-changed
  run: |
    ...
```